modules/photo/test/test_hdr.cpp

```diff
@@ -249,4 +249,22 @@ TEST(Photo_CalibrateRobertson, regression)
     checkEqual(expected, response, 1e-1f, "CalibrateRobertson");
 }
 
+TEST(Photo_CalibrateRobertson, bug_18180)
+{
+    vector<Mat> images;
+    vector<cv::String> fn;
+    glob(string(cvtest::TS::ptr()->get_data_path()) + "hdr/exposures/bug_18180/*.jpg", fn, false);
+    for (auto & i : fn)
+        images.push_back(imread(i));
+    vector<float> times {15, 2.5, 0.25, 0.333333};
+    Mat response, expected;
+    Ptr<CalibrateRobertson> calibrate = createCalibrateRobertson(2, 0.01);
+    calibrate->process(images, response, times);
+    Mat response_no_nans = response.clone();
+    patchNaNs(response_no_nans);
+    // since there should be no NaNs, original response vs. response with NaNs patched should be identical
+    bool isEqual = (sum(response != response_no_nans) == Scalar(0,0,0));
```

Comment:
`norm(response, response_no_nans, NORM_INF) == 0.0` ?

Reply:
That's smarter - let me do that. Thanks!

Reply:
NORM_INF doesn't properly handle NaNs.

There is still a problem:

```
EXPECT_EQ(0.0, cv::norm(response, response_no_nans, NORM_L2));
```

Results to this error message (Linux Debug):

```
Expected equality of these values:
  0.0
    Which is: 0
  cv::norm(response, response_no_nans, NORM_L2)
    Which is: -nan
[  FAILED  ] Photo_CalibrateRobertson.bug_18180 (1789 ms)
```

Reply:
Ah I figured out the issue, I had to move the epsilon back. Thank you!